h2o-r/h2o-package/R/admissibleml.R

```diff
@@ -0,0 +1,150 @@
+#' @param favorable_class Positive/favorable outcome class of the response.
+#' @return frame containing aggregations of intersectional fairness across the models
+#' @export
+h2o.disparate_analysis <-
+  function(models,
+           newdata,
+           protected_cols,
+           reference,
+           favorable_class) {
+    models_info <- .process_models_or_automl(
+        models,
+        newdata,
+        check_x_y_consistency = FALSE,
+        require_multiple_models = TRUE
+      )
+    y <- models_info$y
+    return(cbind(
+      .create_leaderboard(models_info, newdata, top_n = Inf),
+      t(sapply(models, function(model) {
+        capture.output({
+          dm <-
+            h2o.calculate_fairness_metrics(
+              model = model,
+              frame = newdata,
+              protected_cols = protected_cols,
+              reference = reference,
+              favorable_class = favorable_class
+            )$overview
+        })
+        return(
+          c(
+            var = var(dm[["accuracy"]], na.rm = TRUE),
+            corrected_var = .get_corrected_variance(dm),
+            air_min = min(dm$AIR_selectedRatio, na.rm = TRUE),
+            air_mean = mean(dm$AIR_selectedRatio, na.rm = TRUE),
+            air_median = stats::median(dm$AIR_selectedRatio, na.rm = TRUE),
+            air_max = max(dm$AIR_selectedRatio, na.rm = TRUE),
+            cair = stats::weighted.mean(dm$AIR_selectedRatio, dm$relativeSize, na.rm = TRUE),
+            `p.value_min` = min(dm[["p.value"]], na.rm = TRUE),
+            `p.value_median` = stats::median(dm[["p.value"]], na.rm = TRUE),
+            `p.value_mean` = mean(dm[["p.value"]], na.rm = TRUE),
+            `p.value_max` = max(dm[["p.value"]], na.rm = TRUE)
+          )
+        )
+      }))
+    ))
+  }
+
+
+.get_model_list <-
+  function(
+    model_fun,
+    x,
+    y,
+    training_frame,
+    ...
+  ) {
+    obj <- model_fun(x = x, y = y, training_frame = training_frame, ...)
+    models_info <- .process_models_or_automl(obj, train, check_x_y_consistency=FALSE)
+    lapply(models_info$model_ids, models_info$get_model)
+  }
+
+#' Train a grid of models over subsets selected using infogram
+#'
+#' @param ig Infogram object trained with the same protected columns
+#' @param model_fun Function that creates models. This can be something like h2o.automl, h2o.gbm, etc.
+#' @param training_frame Training frame
+#' @param test_frame Test frame
+#' @param y Response column
+#' @param protected_columns Protected columns
+#' @param reference List of values corresponding to a reference for each protected columns.
+#'                  If set to NULL, it will use the biggest group as the reference.
+#' @param favorable_class Positive/favorable outcome class of the response.
+#' @param ... Parameters that are passed to the model_fun.
+#' @return frame containing aggregations of intersectional fairness across the models
+#'
```

Comment:
Can we add an example in the docstring?  I realize it will be a bit long... but maybe it would be helpful?